### ISSUE TITLE
"Save world as..." silently fails if login name is unknown

### ISSUE THREAD

**Issue by aranlunzer** [LivelyKernel/LivelyKernel#336]
Issue world: http://lively-web.org/issues/16-05-25_%22Save_world_as...%22_silently_fails_if_login_name_is_unknown.html

reported by AranLunzer:

When logged in under an ID that doesn't have an associated user directory on lively-web.org (say, "mynickname"), using "Save world as..." and supplying a fully specified path such as  users/SomeLegitName/w.html, nothing appears to happen.

And indeed, the world is not saved.  With no error message, no warning.

**Comment by aranlunzer** [comment]
In case of trouble re-creating:

1. close any browser Incognito windows you have open, and open a new one
2. world menu -> Preferences -> Show login info.  Confirm shown as "unknown_user"
3. "switch user", enter nonsense user name (for me, "aran" was nonsense enough :)
4. world menu -> Save world as...   Enter supposedly valid path (e.g., users/AranLunzer/test123.html)

...and wait.  But not for ever.

**Comment by mroeder** [comment]
I produced a fix on http://lively-web.org/ that is uncommitted:
```diff
diff --git a/core/lively/morphic/Serialization.js b/core/lively/morphic/Serialization.js
index 36e2d5e..6289ca3 100644
--- a/core/lively/morphic/Serialization.js
+++ b/core/lively/morphic/Serialization.js
@@ -229,8 +229,13 @@ lively.morphic.World.saveWorldAs(
             var msg = answer.redirect ?
               "Cannot save as " + url + "\n Save as " + answer.value + " instead?" :
               "You are not allowed to save this world as \n" + url;
-            thenDo && thenDo(new Error(msg));
-            return msg;
+            $world.confirm(msg, function (input) {
+              if (!input)
+                thenDo && thenDo(new Error(msg));
+              else
+                $world.saveWorldAs(answer.value, checkForOverwrites, bootstrapModuleURL, thenDo);
+              });
+            return;
           }
         }
```
@rksm

**Comment by rksm** [comment]
fixed by dac4b1cecc45b7ffb14288b8a3ac255a5c728ad7
